### DISCUSSION TITLE
Embedding resources to compiled executable

### DISCUSSION THREAD

**Discussion by brightening-eyes** [0xc0de/AngieEngine#19] [Q&A] [answered]
[cmrc](https://github.com/vector-of-bool/cmrc) is a resource compiler which can embed resources in the compiled executable, while preserving the filesystem structure.
it would be cool if hork's resource manager adds support for it.
this would add the ability of resource protection in the game at least a little bit.
also, it makes the destribution easier as well.

**Comment by 0xc0de** [comment, marked as answer]
> [cmrc](https://github.com/vector-of-bool/cmrc) is a resource compiler which can embed resources in the compiled executable, while preserving the filesystem structure. it would be cool if hork's resource manager adds support for it. this would add the ability of resource protection in the game at least a little bit. also, it makes the destribution easier as well.

Embedding resources is already available in Hork. Just put the required resource in the "Embedded" folder.

To open a resource, start the file path with the /Embedded/ prefix. For example:
resourceManager->CreateResourceFromFile <MeshResource> ("/Embedded/path/to/file");

**Comment by brightening-eyes** [comment]
hello.
Do I need to put my resources into Embedded directory in hork-source\Engine\Embedded, or the cmake script will detect it and based on it it will add it into the given executable?
also, what if I want to have embedded resources for different custom targets? if I put the resources into embedded directory, in hork-source\Engine\Embedded it will copy out all the resources as very large array blobs which at least makes the executables a little bit larger.
It should be possible in cmake the ability of specifying different directories as embedded ones and for each target.

**Reply by 0xc0de** [reply]
> Do I need to put my resources into Embedded directory in hork-source\Engine\Embedded,

Hello. Yes, you need to place embeddable resources in this directory, cmake will detect them and run the utility to package the resources into a zip archive, which will be embedded as a binary blob into the executable.

> It should be possible in cmake the ability of specifying different directories as embedded ones and for each target.

There is currently no ability to specify different directories for each target. I think that implementing this will not be a problem without using cmrc. I'll create an issue.
